src/App.jsx

```diff
@@ -14,33 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SortType = {
+  ALPHABET: 'alphabet',
+  LENGTH: 'length',
+  DEFAULT: 'default',
+};
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+const handleSort = (type, setGoods, isReversed) => {
+  let sortedGoods = [...goodsFromServer];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  switch (type) {
+    case SortType.ALPHABET:
+      sortedGoods = sortedGoods.sort();
+      break;
+    case SortType.LENGTH:
+      sortedGoods = sortedGoods.sort((a, b) => a.length - b.length);
+      break;
+    default:
+      sortedGoods = [...goodsFromServer];
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  if (isReversed) {
+    sortedGoods.reverse();
+  }
+
+  setGoods(sortedGoods);
+};
+
+const handleReverse = (setGoods, setIsReversed) => {
+  setGoods(prevGoods => [...prevGoods].reverse());
+  setIsReversed(prevIsReversed => !prevIsReversed);
+};
```

Comment:
- You reverse goods in `handleSort`, so you don't need this one
- You don't need update the goods state. It could be just a variable